### ISSUE TITLE
NewsBuilder should add more space between sections

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4291]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#4291|
|Type|enhancement|
|Created|2010-02-22 02:51:58Z|
|Branch|https://github.com/twisted/twisted/tree/news-builder-lint-4291|

Projects are only separated by one line, it should be two.

Attachments:

* [twisted-python-release-newlines.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a86/a86ea47254b1956667196a96b5b832a876c8bfca/944deb726e3d594943dbec50af0f495c74c9ff7c.diff) (1562 bytes) - added by djfroofy on 2010-02-22 14:46:21Z - Replace projects with two newlines
* [twisted-python-release-omitEmptyProjects.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a86/a86ea47254b1956667196a96b5b832a876c8bfca/124b2226552cc2eeefc1b9a6a583ba24bedf9a05.diff) (2521 bytes) - added by djfroofy on 2010-02-22 14:47:01Z - Omit projects with no changes from NEWS
* [patch-0.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a86/a86ea47254b1956667196a96b5b832a876c8bfca/8015a68ffd3941b2de5dff7af6a48ae1bbae1277.diff) (3310 bytes) - added by djfroofy on 2010-02-22 16:41:22Z - Rebuilt patch including newline and project omission

<details><summary>Searchable metadata</summary>

```
trac-id__4291 4291
type__enhancement enhancement
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_news_builder_lint_4291 branches/news-builder-lint-4291
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__easy easy
time__1266807118000000 1266807118000000
changetime__1266864677000000 1266864677000000
version__None None
owner__ 
cc__khorn
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

Also, it should probably not mention projects without any changes, instead of displaying an empty section.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)|@djfroofy set owner to @djfroofy<br>@djfroofy set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_3"></a>|@djfroofy commented|
|-|-|

Since this is really two isolated changes, I feel like this would be cleaner done in a branch.  But I don't have commit privileges - such would be nice.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_4"></a>|@djfroofy commented|
|-|-|

For now, I'm attaching two patches: 1) separating projects with double newline; 2) omitting projects with no changes.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve<br>@therve set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve commented|
|-|-|

(In [28376]) Branching to 'news-builder-lint-4291'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @djfroofy|
|-|-|

Thanks for the patch! I think the first change is trivial enough that we can do it at the same time.

For the second patch, we shouldn't computer the NEWS data 2 times in build, but reuse them once we detected that it's not empty. Everything else looks great.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_8"></a>|@djfroofy set status to `assigned`|
|-|-|

That's odd. I think some lines might have been rejected somehow when you applied the patch - I was computing NEWS only once and reusing in build(). I'll fix failed patch now.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)|@djfroofy set owner to @therve<br>@djfroofy set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set status to `closed`|
|-|-|

(In [28402]) Merge news-builder-lint-4291

Author: djfroofy
Reviewer: therve
Fixes [#4291](https://github.com/twisted/twisted/issues/4291)

The NEWSBuilder now adds some space between project sections, and doesn't
display a section for a project without changes.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
